### PR TITLE
SAK-38240 Students can access to assignments list bug fixed

### PR DESCRIPTION
**Pull request by druiz309**
This PR is for fixing an important issue that allows students to view assignments list from lessons (in students_view_grade) even though assignments tool is not visible.

### CODE REVIEW

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 1]
```diff
@@ -2706,6 +2706,26 @@ protected String build_student_view_grade_context(VelocityPortlet portlet, Conte
             context.put("itemHelpers", itemHelpers);
             context.put("taggable", Boolean.valueOf(true));
         }
+        
+        try {
+            // Get site ID
+            String siteId = toolManager.getCurrentPlacement().getContext();
+            Optional<Site> site = Optional.of(siteService.getSite(siteId));
```
Comment (mpellicer) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java:2713]: Could you not use the Optional pattern here?

File: assignment/tool/src/webapp/vm/assignment/chef_assignments_student_view_grade.vm [hunk 2]
```diff
@@ -389,9 +389,11 @@
     </div>
     <form action="#toolForm("AssignmentAction")" method="post">
         <div class="act">
-            <input type="submit" accesskey="x" name="eventSubmit_doCancel_view_grade"
-                   value="$tlang.getString("gen.backtolist")" accesskey="x" class="active TB_hideControl"
-                   onclick="SPNR.disableControlsAndSpin( this, null );"/>
+            #if ($isAssignmentsToolVisible)
```
Comment (mpellicer) [inline assignment/tool/src/webapp/vm/assignment/chef_assignments_student_view_grade.vm:392]: Maybe this should be applied at FORM level instead to button?

**Comment by druiz309** [comment]
I have just made the changes

**Comment by mpellicer** [comment]
WDYT @josecebe !

**Review by mpellicer — APPROVED** [review]
Thanks!

**Pull request merged**